### DISCUSSION TITLE
Introduce strict (and unified) code style and formatting

### DISCUSSION THREAD

**Discussion by iakov** [MeVisLab/pythonqt#234] [General]
This can simplify code editing, and PRs will be autoformatted. Code formatting can be automated via editors and cli-tools, also checks can be added.
I suggest to use `clang-format`-based automation. Qt Creator supports it too.
Some (any) style can be chosen (f.e., [from KDevelop](https://github.com/KDE/kdevelop/blob/master/.clang-format)).
If it is welcomed by the maintainers and community, we (my team) can PR some `.clang-format` config with corresponding CI config to check PRs, and maybe other automation features (f.e. pre-commit).

**Comment by mrbean-bremen** [comment]
I'm all for it. I don't want to think about formatting, and don't like to read code that is formatted differently in different places. `clang-format` also sounds good, as it is supported by most editors, so from my POV, I would be happy to see such a PR!

**Reply by usiems** [reply]
Should this only apply to the PythonQt library itself, or also to the generator? The generator is composed of sources from different sources, with vastly different styles, I don't know how much manual work it would require to unify this code.

**Reply by iakov** [reply]
I suggest unifying formatting for all the source, not considering advanced linting like naming convention check ATM. Tools like clang-format or [uncrustify](https://github.com/uncrustify/uncrustify) can reformat the code according to rules, so there is no obvious reason for manual work. The only thing to do carefully for the first time is to choose format options as close to the current style as possible. 
We are playing ATM in our fork, and the PR to your upstream will be done later.
